[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #160611
* #160601
* #160470
* #159786
* #160138
* #159817
* #159678
* #159329
* #144516
* #159281
* __->__ #157971

The comments/conventions are not exactly correct here, as the implementation at this PR is partial. They will be fixed in #160138.

No tests added, since there shouldn't be any overall semantic changes.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

williamwen42 committed:
[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 reviewed (approved):
I spent around 10 min trying to understand the details, but did not understand everything. The comments make sense, but I don't have all the info to check each line. 

Given that this is fundamental, I am relying on tests for surity.

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"


The comments/conventions are not exactly correct here, as the implementation at this PR is partial. They will be fixed in #160138.

No tests added, since there shouldn't be any overall semantic changes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] implement new resume frame stack/locals/cell layout convention"


The comments/conventions are not exactly correct here, as the implementation at this PR is partial. They will be fixed in #160138.

No tests added, since there shouldn't be any overall semantic changes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
